samples/ipc_call_interface/inc/math_tlb.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include "az_ulib_ipc_api.h"
+#include "ulib_result.h"
+#include <stdint.h>
+
+/*
+ * interface definition
+ */
+#define MATH_INTERFACE_NAME "math"
+#define MATH_INTERFACE_VERSION 1
+typedef az_ulib_ipc_interface_handle math_handle;
+
+static inline AZ_ULIB_RESULT math_create(math_handle handle) {
+  return az_ulib_ipc_try_get_interface(
+      MATH_INTERFACE_NAME, MATH_INTERFACE_VERSION, AZ_ULIB_VERSION_EQUALS_TO, handle);
+}
+
+static inline void math_destroy(math_handle handle) { az_ulib_ipc_release_interface(handle); }
+
+/*
+ * Define sum method on math interface.
+ */
+#define MATH_INTERFACE_SUM_METHOD_NAME "sum"
+#define MATH_INTERFACE_SUM_METHOD 0
+
+typedef struct SUM_MODEL_IN_TAG {
+  int32_t a;
+  int32_t b;
+} SUM_MODEL_IN;
+
+typedef int64_t SUM_MODEL_OUT;
+
+/*
+ * Inter-component Calling Wrapper for sum.
+ */
+static inline AZ_ULIB_RESULT sum(math_handle handle, int32_t a, int32_t b, int64_t *res) {
+  SUM_MODEL_IN sum_in;
+  SUM_MODEL_OUT sum_out;
+
+  // Marshalling
+  sum_in.a = a;
+  sum_in.b = b;
+
+  // Call
+  AZ_ULIB_RESULT result = az_ulib_ipc_call(handle, MATH_INTERFACE_SUM_METHOD, &sum_in, &sum_out);
+
+  // Unmarshalling
+  *res = sum_out;
+
+  return result;
+}
+
+/*
+ * Define sub method on math interface.
+ */
+#define MATH_INTERFACE_SUB_METHOD_NAME "sub"
+#define MATH_INTERFACE_SUB_METHOD 1
```

Comment:
>MATH_INTERFACE_SUB_METHOD [](start = 8, length = 25)

should we think about a way for these to be generated via an enum instead of by hand. what happens if someone doesn't increment their #defines correctly?

Reply:
This whole file shall be autogenerated. I don't like it created by hand as well, but I didn't find a better way to do that.

---
In reply to: [338283425](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338283425) [](ancestors = 338283425)

Reply:
Moving to enum.

---
In reply to: [338310590](https://github.com/Azure/azure-ulib-c/pull/40#discussion_r338310590) [](ancestors = 338310590,338283425)